### ISSUE TITLE
Targets file contains actions for build target, but not for publish

### ISSUE THREAD

**Issue by Aleksandar-1** [hms-networks/ixxat-vci4-dotnet-wrapper#2]
Hello,

we are consuming the Ixxat nuget package from nuget.org:
https://www.nuget.org/packages/Ixxat.Vci4

When we run our build pipeline with
dotnet build
the files vcinet.x86.dll and Ijwhost.dll are copied to the correct location in the build output folder ($OutDir).

When we run our build pipeline with
dotnet publish
we notice, that the files are missing in the publish folder ($PublishDir). That is because in the targets file Ixxat.Vci4.targets, only the build target is declared:

```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Target Name="CopyNativeLibs_x86" AfterTargets="Build">
    <ItemGroup>
      <NativeLibs_x86    Include="$(MSBuildThisFileDirectory)\x86\vcinet.x86.dll" />
      <DotNetHostLib_x86 Include="$(MSBuildThisFileDirectory)\x86\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x86)"    DestinationFolder="$(OutDir)\vcinet\x86" />
    <Copy SourceFiles="@(DotNetHostLib_x86)" DestinationFolder="$(OutDir)\vcinet\x86" />
  </Target>
  <Target Name="CopyNativeLibs_x64" AfterTargets="Build">
    <ItemGroup>
      <NativeLibs_x64    Include="$(MSBuildThisFileDirectory)\x64\vcinet.x64.dll" />
      <DotNetHostLib_x64 Include="$(MSBuildThisFileDirectory)\x64\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x64)"    DestinationFolder="$(OutDir)\vcinet\x64" />
    <Copy SourceFiles="@(DotNetHostLib_x64)" DestinationFolder="$(OutDir)\vcinet\x64" />
  </Target>
</Project>
```

It is possible though to add handling for the target publish. Notice that the whole target has to be declared new for the publish target, since, the output folder must be $PublishDir instead of $OutDir. For instance:

```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Target Name="CopyNativeLibs_x86" AfterTargets="Build">
    <ItemGroup>
      <NativeLibs_x86    Include="$(MSBuildThisFileDirectory)\x86\vcinet.x86.dll" />
      <DotNetHostLib_x86 Include="$(MSBuildThisFileDirectory)\x86\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x86)"    DestinationFolder="$(OutDir)\vcinet\x86" />
    <Copy SourceFiles="@(DotNetHostLib_x86)" DestinationFolder="$(OutDir)\vcinet\x86" />
  </Target>
  <Target Name="CopyNativeLibs_x64" AfterTargets="Build">
    <ItemGroup>
      <NativeLibs_x64    Include="$(MSBuildThisFileDirectory)\x64\vcinet.x64.dll" />
      <DotNetHostLib_x64 Include="$(MSBuildThisFileDirectory)\x64\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x64)"    DestinationFolder="$(OutDir)\vcinet\x64" />
    <Copy SourceFiles="@(DotNetHostLib_x64)" DestinationFolder="$(OutDir)\vcinet\x64" />
  </Target>

  <Target Name="CopyNativeLibs_x86" AfterTargets="Publish">
    <ItemGroup>
      <NativeLibs_x86    Include="$(MSBuildThisFileDirectory)\x86\vcinet.x86.dll" />
      <DotNetHostLib_x86 Include="$(MSBuildThisFileDirectory)\x86\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x86)"    DestinationFolder="$(PublishDir)\vcinet\x86" />
    <Copy SourceFiles="@(DotNetHostLib_x86)" DestinationFolder="$(PublishDir)\vcinet\x86" />
  </Target>
  <Target Name="CopyNativeLibs_x64" AfterTargets="Publish">
    <ItemGroup>
      <NativeLibs_x64    Include="$(MSBuildThisFileDirectory)\x64\vcinet.x64.dll" />
      <DotNetHostLib_x64 Include="$(MSBuildThisFileDirectory)\x64\Ijwhost.dll" />
    </ItemGroup>
    <Copy SourceFiles="@(NativeLibs_x64)"    DestinationFolder="$(PublishDir)\vcinet\x64" />
    <Copy SourceFiles="@(DotNetHostLib_x64)" DestinationFolder="$(PublishDir)\vcinet\x64" />
  </Target>
</Project>
```

Do you think, you can incorporate this into the Ixxat nuget package? It would simplify the use of dotnet publish.

This concerns the packages:
https://www.nuget.org/packages/Ixxat.Vci4
https://www.nuget.org/packages/Ixxat.Vci4.StrongName

Best Regards,
Aleksandar

**Comment by HMS-klob** [comment]
I think it's a reasonable change and I will look into this ASAP.

**Comment by HMS-klob** [comment]
Thank you for reporting. Fixed in version 4.1.3, package upload complete, should be available after indexing.

**Comment by Aleksandar-1** [comment]
Thank you very much for the quick response!

**Comment by Aleksandar-1** [comment]
Hello,

we have noticed the following:
If the targets which have the same name, only the last definition will be used. This is described in: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-targets?view=vs-2022#declare-targets-in-the-project-file

![grafik](https://github.com/hms-networks/ixxat-vci4-dotnet-wrapper/assets/95416095/aa350d65-8ccc-4d3f-a010-41aabb784b2d)

That means, that after adding the publish targets, the target names exist two times, and only the second definition will be used.
For instance:

CopyNativeLibs_x86 exists one time for build, and one time for publish. (also true for CopyNativeLibs_x64).
But only the publish definition will be used. That can be verified, by executing one time the build and one time the publish target. At the moment, when executing the "build" target, nothing will be copied, only on "publish".

The fix to this is, to make the names of the targets unique, for instance adding the action to the target name:

`<Target Name="CopyNativeLibs_x86" AfterTargets="Build">` gets `<Target Name="CopyNativeLibs_x86_Build" AfterTargets="Build">`

`<Target Name="CopyNativeLibs_x64" AfterTargets="Build">` gets `<Target Name="CopyNativeLibs_x64_Build" AfterTargets="Build">`

`<Target Name="CopyNativeLibs_x86" AfterTargets="Publish">` gets `<Target Name="CopyNativeLibs_x86_Publish" AfterTargets="Publish">`

`<Target Name="CopyNativeLibs_x64" AfterTargets="Publish">` gets `<Target Name="CopyNativeLibs_x64_Publish" AfterTargets="Publish">`

etc .....


Regards,
Aleksandar

**Comment by HMS-klob** [comment]
Ok, fixed in version 4.1.4, upload in progress.

**Comment by Aleksandar-1** [comment]
Thank you

Regards,
Aleksandar
